armi/physics/fuelCycle/fuelHandlers.py

```diff
@@ -1460,29 +1461,78 @@ def _swapFluxParam(self, incoming, outgoing):
         Assumes assemblies have the same block structure. If not, blocks will be swapped one-for-one until
         the shortest one has been used up and then the process will truncate.
         """
-        if len(incoming) != len(outgoing):
-            runLog.warning(
-                "{0} and {1} have different numbers of blocks. Flux swapping (for XS weighting) will "
-                "be questionable".format(incoming, outgoing)
+        # Find the block-based mesh points for each assembly
+        meshIn = self.r.core.findAllAxialMeshPoints([incoming], False)
+        meshOut = self.r.core.findAllAxialMeshPoints([outgoing], False)
+
+        # If the assembly mesh points don't match, the swap won't be easy
+        if meshIn != meshOut:
+            runLog.debug(
+                "{0} and {1} have different meshes, resampling.".format(
+                    incoming, outgoing
+                )
             )
+
+            # grab the current values for incoming and outgoing
+            fluxIn = [b.p.flux for b in incoming]
+            mgFluxIn = [b.p.mgFlux for b in incoming]
+            powerIn = [b.p.power for b in incoming]
+            fluxOut = [b.p.flux for b in outgoing]
+            mgFluxOut = [b.p.mgFlux for b in outgoing]
+            powerOut = [b.p.power for b in outgoing]
+
+            # resample incoming to outgoing, and vice versa
+            fluxOutNew = resampleStepwise(meshIn, fluxIn, meshOut)
+            mgFluxOutNew = resampleStepwise(meshIn, mgFluxIn, meshOut)
+            powerOutNew = resampleStepwise(meshIn, powerIn, meshOut, avg=False)
+            fluxInNew = resampleStepwise(meshOut, fluxOut, meshIn)
+            mgFluxInNew = resampleStepwise(meshOut, mgFluxOut, meshIn)
+            powerInNew = resampleStepwise(meshOut, powerOut, meshIn, avg=False)
+
+            # load the new outgoing values into place
+            for b, flux, mgFlux, power in zip(
+                outgoing, fluxOutNew, mgFluxOutNew, powerOutNew
+            ):
+                b.p.flux = flux
+                b.p.mgFlux = mgFlux
+                b.p.power = power
+                b.p.pdens = power / b.getVolume()
+
+            # load the new incoming values into place
+            for b, flux, mgFlux, power in zip(
+                incoming, fluxInNew, mgFluxInNew, powerInNew
+            ):
+                b.p.flux = flux
+                b.p.mgFlux = mgFlux
+                b.p.power = power
+                b.p.pdens = power / b.getVolume()
+
+            return
+
+        # Since the axial mesh points match, do the simple swap
         for bi, (bIncoming, bOutgoing) in enumerate(zip(incoming, outgoing)):
```

Comment:
what if outgoing  length is longer @john-science ? wouldn't some block still be unset?